src/replicache.ts

```diff
@@ -1599,30 +1617,49 @@ export class Replicache<MD extends MutatorDefs = {}> {
         return;
       }
 
-      this._lc.debug?.(
-        `Client ${selfClientID} recovered mutations for client ` +
-          `${clientID}.  Details`,
-        {
-          mutationID: client.mutationID,
-          lastServerAckdMutationID: client.lastServerAckdMutationID,
-          lastMutationID: pullResponse?.lastMutationID,
-        },
-      );
+      if (pullResponse && isClientStateNotFoundResponse(pullResponse)) {
+        this._lc.debug?.(
+          `Client ${selfClientID} cannot recover mutations for client ` +
+            `${clientID}. The client no longer exists on the server.`,
+        );
+      } else {
+        this._lc.debug?.(
+          `Client ${selfClientID} recovered mutations for client ` +
+            `${clientID}.  Details`,
+          {
+            mutationID: client.mutationID,
+            lastServerAckdMutationID: client.lastServerAckdMutationID,
+            lastMutationID: pullResponse?.lastMutationID,
+          },
+        );
+      }
       const newClientMap = await persist.updateClients(
         (clients: persist.ClientMap) => {
           assertNotUndefined(pullResponse);
+
           const clientToUpdate = clients.get(clientID);
+          if (isClientStateNotFoundResponse(pullResponse)) {
+            if (!clientToUpdate) {
+              return persist.noClientUpdates;
+            }
+
+            const newClients = new Map(clients);
```

Comment:
This is basically an early GC of the client.  I guess this is the best thing to do as these mutations will never be successfully recovered.  Wondering if we should log about this at some higher level than debug.